### PR TITLE
Export API keys for search service

### PR DESCRIPTION
**Pull request by super-harsh**
**What this PR does / why we need it**:

Closes: #3053 

Adding the support for exporting API keys for Azure Cognitive Search and reviving the `authOptions` property to enable users to select aadOrApiKey auth mode. 

**Special notes for your reviewer**:

Also had to add the custom key matcher logic to replace/hide the base64 keys without '=' padding for the returned API keys

**If applicable**:
- [x] this PR contains tests

### CODE REVIEW

File: v2/api/search/customizations/search_service_extension.go [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+	for _, key := range keys {
+		if key == nil {
+			continue
+		}
+
+		// We have to do it this way, since the autogenerated query key has key.Name == nil. See screenshot in https://learn.microsoft.com/en-us/azure/search/search-security-api-keys?tabs=portal-use%2Cportal-find%2Cportal-query#find-existing-keys
+		if key.Name == nil && key.Key != nil {
+			result["default"] = *key
+			continue
+		}
```
Comment (theunrepentantgeek) [inline v2/api/search/customizations/search_service_extension.go:128]: Blank line after closing `}` #nit

**Review by theunrepentantgeek — CHANGES_REQUESTED** [review]
Changes look good for exposing the keys, but how are you addressing the ARM API shape required for `apiKeyOnly` access?

The required ARM payload is this:

``` yaml
"authOptions": {
    "apiKeyOnly": {}
}
```

I don't see any changes to the ARM types or the CRD shape in this PR to address this.

**Review by theunrepentantgeek — APPROVED** [review]
Looks good.

File: v2/samples/search/v1api/v1api20220901_searchservice.yaml [hunk 2]
```diff
@@ -17,3 +17,17 @@ spec:
...
+    secrets:
+      adminPrimaryKey:
+        key: adminPrimaryKey
+        name: searchservicekeyssecret
+      adminSecondaryKey:
+        key: adminSecondaryKey
+        name: searchservicekeyssecret
+      queryKey:
+        key: queryKey
+        name: searchservicekeyssecret
```
Comment (matthchr) [inline v2/samples/search/v1api/v1api20220901_searchservice.yaml:33]: minor: should have a newline at the end here

**Review by matthchr — APPROVED** [review]

**Pull request merged**